### PR TITLE
Tidy up code

### PR DESCRIPTION
**Pull request by michaelcspeed**
Made some improvements to readability.  @osamaasifoak lemme know if they make sense to you

### CODE REVIEW

File: lib/providers/player/player_provider.dart [hunk 1]
```diff
@@ -14,100 +14,99 @@ final playerProvider =
...
     required TrackFilesModel file,
   }) async {
-    var _track = trackModel.customCopyWith();
-    for (var i = 0; i < _track.audio.length; i++) {
-      var element = _track.audio[i];
-      var fileIndex = element.files.indexWhere((e) => e.id == file.id);
+    var track = trackModel.customCopyWith();
+
+    track.audio.forEach((audioModel) {
+      var fileIndex = audioModel.files.indexWhere((it) => it.id == file.id);
```
Comment (osamaasifoak) [inline lib/providers/player/player_provider.dart:25]: what's the meaning of **it**?

File: lib/providers/player/audio_player_provider.dart [hunk 2]
```diff
@@ -19,10 +20,11 @@ final audioPlayerNotifierProvider =
 class AudioPlayerNotifier extends BaseAudioHandler
     with QueueHandler, SeekHandler, ChangeNotifier {
   var backgroundSoundAudioPlayer = AudioPlayer();
-  TrackFilesModel? currentlyPlayingTrack;
-  final hasBgSound = 'hasBgSound';
   final trackAudioPlayer = AudioPlayer();
-  final fadeDuration = 5;
+
+  TrackFilesModel? currentlyPlayingTrack;
+  final hasBgSoundId = 'hasBgSound';
```
Comment (osamaasifoak) [inline lib/providers/player/audio_player_provider.dart]: I think the name should be hasBgSound instead of hasBgSoundId. Lmk, what do you think?

**Review by osamaasifoak — APPROVED** [review]
Hey @michaelcspeed, I'm approving this PR. The changes you have made are looking good to me thanks man :)

File: lib/providers/player/audio_player_provider.dart (hunk 2 above)
Comment (michaelcspeed) [reply lib/providers/player/audio_player_provider.dart]: `hasBGSound` implies it's a boolean, but this is really a key... maybe it should be called `hasBgSoundKey`

File: lib/providers/player/player_provider.dart (hunk 1 above)
Comment (michaelcspeed) [reply lib/providers/player/player_provider.dart:25]: It's the same as a for loop, but directly gives you the item instead if incrementing an index

File: lib/providers/player/audio_player_provider.dart (hunk 2 above)
Comment (michaelcspeed) [reply lib/providers/player/audio_player_provider.dart]: Also 'trackId' and 'fileId' should be constants too but I didn't do them here

**Comment by michaelcspeed** [comment]
@osamaasifoak If you're happy with this, please merge it once you've resolved the comments. Thanks!

File: lib/providers/player/audio_player_provider.dart (hunk 2 above)
Comment (osamaasifoak) [reply lib/providers/player/audio_player_provider.dart]: Correct **hasBgSound** should be called **hasBgSoundKey**

**Pull request merged**